lib/src/store/moor/moor_store.dart

```diff
@@ -397,6 +417,152 @@ class MoorStore extends Store {
+          .where((element) => element.roomId?.value != null)
+          .map((e) => e.roomId!.value)
+          .toList(),
+    );
+
+    final List<Room> roomsResult = [];
+    for (final record in roomRecords) {
+      final roomRecord = record.roomRecord;
+      final roomId = roomRecord.id;
+      final roomContext = context != null
+          ? RoomContext.inherit(
+              context,
+              roomId: RoomId(roomId),
+            )
+          : null;
+
+      //Make Timeline with messagesEventFiltered from messageEventMap and messageEventResult
+      final messagesIndexes = messageEventMap[roomId] ?? [];
+      final List<RoomEvent> messagesEventFiltered = [];
+      for (final i in messagesIndexes) {
+        messagesEventFiltered.add(messageEventResult[i]);
+      }
+      final Timeline? timeline = Timeline(
+        messagesEventFiltered,
+        context: roomContext,
+        previousBatch: roomRecord.timelinePreviousBatch,
+        previousBatchSetBySync: roomRecord.timelinePreviousBatchSetBySync,
+      );
+
+      //Make MemberTimeline with messageMembersFiltered from messagesMembersMap and messageMembers
+      final membersIndexes = messagesMembersMap[roomId] ?? [];
+
+      final List<Member> messageMembersFiltered = [];
+      for (final i in membersIndexes) {
+        messageMembersFiltered.add(messageMembers[i]);
+      }
+
+      final MemberTimeline? memberTimeline = MemberTimeline(
+        messageMembersFiltered,
+        context: roomContext,
+      );
+
+      //Make Ephemeral with ephemeralEventsFiltered from ephemeralMap and ephemeralEventResult
+      final epernalEventsIndexes = ephemeralMap[roomId] ?? [];
+      final List<EphemeralEvent> ephemeralEventsFiltered = [];
+      for (final i in epernalEventsIndexes) {
+        ephemeralEventsFiltered.add(ephemeralEventResult[i]);
+      }
+      final ephemeral = Ephemeral(ephemeralEventsFiltered);
+
+      final room = Room(
+        context: context,
+        id: RoomId(roomRecord.id),
+        stateEvents: RoomStateEvents(
+          nameChange: record.nameChangeRecord?.toRoomEvent(),
+          avatarChange: record.avatarChangeRecord?.toRoomEvent(),
+          topicChange: record.topicChangeRecord?.toRoomEvent(),
+          powerLevelsChange: record.powerLevelsChangeRecord?.toRoomEvent(),
+          joinRulesChange: record.joinRulesChangeRecord?.toRoomEvent(),
+          canonicalAliasChange:
+              record.canonicalAliasChangeRecord?.toRoomEvent(),
+          creation: record.creationRecord?.toRoomEvent(),
+          upgrade: record.upgradeRecord?.toRoomEvent(),
+        ),
+        timeline: timeline,
+        memberTimeline: memberTimeline,
+        summary: RoomSummary(
+          joinedMembersCount: roomRecord.summaryJoinedMembersCount,
+          invitedMembersCount: roomRecord.summaryInvitedMembersCount,
+        ),
+        directUserId: roomRecord.directUserId != null
+            ? UserId(roomRecord.directUserId ?? '')
```

Comment:
Why do we use ?? check, if we have already checked, that value != null?

Reply:
FIxed